### PR TITLE
replace-ad_rw

### LINKED ISSUE #248: Replace `ad_rw` with `epics_signal_rw_rbv`
Does it make sense to update the signals in AD with this e.g. https://github.com/bluesky/ophyd-async/blob/c42afcb1f801d5f3f6ae88fd6a7fa8e0c7043893/src/ophyd_async/epics/areadetector/utils.py#L11?

_Originally posted by @DominicOram in https://github.com/bluesky/ophyd-async/issues/221#issuecomment-2066194461_

### PR DESCRIPTION
**Pull request by ZohebShaikh**
closes  #248

### CODE REVIEW

**Comment by ZohebShaikh** [comment]
I think we can replace ad_r as well 
```
def ad_r(datatype: Type[T], prefix: str) -> SignalR[T]:
    return epics_signal_r(datatype, prefix + "_RBV")
```
it just calls to epics_signal_r internally

**Comment by coretl** [comment]
> I think we can replace ad_r as well
> 
> ```
> def ad_r(datatype: Type[T], prefix: str) -> SignalR[T]:
>     return epics_signal_r(datatype, prefix + "_RBV")
> ```
> 
> it just calls to epics_signal_r internally

Yes please

**Review by coretl — COMMENTED** [review]
Looks good, please can you also replace `ad_r` and then delete the `utils.py` that contains it

File: src/ophyd_async/epics/areadetector/drivers/ad_base.py [hunk 1]
```diff
@@ -43,14 +43,14 @@ class DetectorState(str, Enum):
...
-        self.array_size_x = ad_r(int, prefix + "ArraySizeX")
-        self.array_size_y = ad_r(int, prefix + "ArraySizeY")
-        self.detector_state = ad_r(DetectorState, prefix + "DetectorState")
+        self.acquire = epics_signal_rw_rbv(bool, prefix + "Acquire")
+        self.acquire_time = epics_signal_rw_rbv(float, prefix + "AcquireTime")
+        self.num_images = epics_signal_rw_rbv(int, prefix + "NumImages")
+        self.image_mode = epics_signal_rw_rbv(ImageMode, prefix + "ImageMode")
+        self.array_counter = epics_signal_rw_rbv(int, prefix + "ArrayCounter")
+        self.array_size_x = epics_signal_r(int, prefix + "ArraySizeX")
+        self.array_size_y = epics_signal_r(int, prefix + "ArraySizeY")
```
Comment (coretl) [inline src/ophyd_async/epics/areadetector/drivers/ad_base.py:52]: I notice that `ad_r` adds the `_RBV` suffix, so every replacement of it should do the same:
```suggestion
        self.array_size_y = epics_signal_r(int, prefix + "ArraySizeY_RBV")
```

**Review by coretl — CHANGES_REQUESTED** [review]

File: src/ophyd_async/epics/areadetector/drivers/ad_base.py [hunk 2]
```diff
@@ -43,14 +43,16 @@ class DetectorState(str, Enum):
...
         # Define some signals
-        self.acquire = ad_rw(bool, prefix + "Acquire")
-        self.acquire_time = ad_rw(float, prefix + "AcquireTime")
-        self.num_images = ad_rw(int, prefix + "NumImages")
-        self.image_mode = ad_rw(ImageMode, prefix + "ImageMode")
-        self.array_counter = ad_rw(int, prefix + "ArrayCounter")
-        self.array_size_x = ad_r(int, prefix + "ArraySizeX")
-        self.array_size_y = ad_r(int, prefix + "ArraySizeY")
-        self.detector_state = ad_r(DetectorState, prefix + "DetectorState")
+        self.acquire = epics_signal_rw_rbv(bool, prefix + "Acquire_RBV")
```
[the remark this answers is not in the document]
Comment (ZohebShaikh) [reply src/ophyd_async/epics/areadetector/drivers/ad_base.py]: I had a look at the function for `epics_signal_rw_rbv` it adds the suffix for it ..I have now removed it from the PR

**Review by coretl — APPROVED** [review]

**Pull request merged**